lib/spack/spack/package.py

```diff
@@ -2162,6 +2172,27 @@ def do_uninstall(self, force=False):
         # delegate to instance-less method.
         Package.uninstall_by_spec(self.spec, force)
 
+    def do_deprecate(self, replacement, link_fn):
+        """Deprecate this package in favor of spec"""
+        # Copy metadata to deprecated directory of replacement
+        spec = self.spec
+        deprecated = spack.store.db.get_record(spec).deprecated_for
+        if deprecated:
+            if deprecated == replacement.dag_hash():
+                s = spec.format('{name}/{hash:7}')
```

Comment:
(question) when would `spec.dag_hash()` not match `replacement.dag_hash()` (unless spec was not concrete)?

Can you deprecate the same spec multiple times?

Reply:
It wouldn't match if you're trying to deprecate a spec that has already been deprecated in favor of a different spec.

You cannot deprecate the same spec multiple times. In a later PR it may be nice to add a feature where you can re-deprecate it to change the replacement spec.